lua/nvim-tree/view.lua

```diff
@@ -89,9 +89,13 @@ function M._wipe_rogue_buffer()
   if bn then
     local win_ids = vim.fn.win_findbuf(bn)
     for _, id in ipairs(win_ids) do
-      a.nvim_win_close(id, true)
+      if vim.fn.win_gettype(id) ~= "autocmd" then
+        a.nvim_win_close(id, true)
+      end
     end
-    a.nvim_buf_delete(bn, {})
+    -- Do not use `nvim_buf_delete` here as it throws an uncacthable error if
+    -- the buffer is in an autocmd window.
+    a.nvim_command("silent! bw " .. bn)
```

Comment:
I guess the reason its not caught by `pcall` is that the window isn't closed directly as a consequence of `nvim_buf_delete`, but vim tries to close it because we delete the last buffer in the window. Maybe upstream bug?

Reply:
did you try to schedule the buf_delete call ?

Reply:
Good call, scheduling seems to work quite well!

Reply:
yep, because it'll send the command in the event loop, thus all the other synchronous stuff must be done before triggering this :)